BootloaderCorePkg/Tools/CfgDataStitch.py

```diff
@@ -82,12 +82,20 @@ def cfgdata_stitch(ifwi_file, ifwi_out_file, cfg_dir, key_file, script_dir, tool
     if not os.path.exists(os.path.join(cfg_dir ,'CfgDataDef.dsc')):
         raise Exception("No CfgDataDef.dsc file found under directory '%s' !" % cfg_dir)
 
-    # ensure all required files exist
-    chk_files = [
-        (script_dir, ['GenCfgData.py', 'CfgDataTool.py']),
-        (os.path.dirname(key_file), [os.path.basename(key_file)]),
-        (cfg_dir, ['CfgDataDef.dsc']),
-    ]
+    # Check for KEY_ID at end of string
+    if (key_file.endswith('KEY_ID')):
+        # ensure all required files exist
+        chk_files = [
+            (script_dir, ['GenCfgData.py', 'CfgDataTool.py']),
+            (cfg_dir, ['CfgDataDef.dsc']),
+        ]
+    else:
+        # ensure all required files exist
+        chk_files = [

```

Comment:
If a HSM is being used to perform signing, won't this check fail ?
Is it ok if we let the execution continue here and let the failure happen in signing script ?

Reply:
Yes for key_file, it can be handled in single signing script.
I would remove key file check from here for real path/key id.

Reply:
@Subash Ignore my original comment.

Reply:
> @Subash Ignore my original comment.

Sorry I can't. Github sent me a notification to make sure I look at it.